### PR TITLE
Proposed flush fix

### PR DESCRIPTION
**Pull request by rob-c**
This is a proposed quick fix for #360. I've not had time to test it but it 'feels' faster from a very unscientific test of submitting 200 subjobs and monitoring them by hand, but this could be the hostname changes from recent changes.

I'm curious to see if the tests are effected by this.

### CODE REVIEW

**Comment by rob-c** [comment]
Ouch, 'climbing down' the ladder appears to be a bit dangerous. I'm seeing some infinite loops when the ```_setFlushed``` trys to climb down the children within the schema.

I'll do some looking at this over the weekend if I get a chance. A bit disappointing as this worked in an isolated test by itself.

**Comment by rob-c** [comment]
OK, I think I've found a fix, lets see if all tests pass with this.

**Comment by rob-c** [comment]
retest this please

**Comment by rob-c** [comment]
I just fixed a bug in the XML test. Basically the subjob XML was only written once even though the subjob went from new to completed in under 30sec (proof that this PR is working).
The fix was to mark all subjobs dirty and trigger a flush of the ```SubJobXMLList``` class to make sure at least 2 write events happened on a per-subjob basis.

This seems to save time (on subjob related tests) so I'm hoping to do some testing to see if monitoring a job with 1k subjobs is now manageable.

**Comment by rob-c** [comment]
Any objections to merging this first thing Monday?

File: python/Ganga/GPIDev/Base/Objects.py [hunk 1]
```diff
@@ -957,12 +957,20 @@ def _getRegistryID(self):
...
+        """ Set the dirty flag all the way up to the parent"""
         self._dirty = True
         parent = self._getParent()
         if parent is not None:
             parent._setDirty()
 
     def _setFlushed(self):
+        """Un-Set the dirty flag all of the way down the schema."""
+        if self._schema:
+            for k in self._schema.allItemNames():
```
Comment (milliams) [inline python/Ganga/GPIDev/Base/Objects.py]: This should only need to be done over the component items.

Comment (rob-c) [inline python/Ganga/GPIDev/Base/Objects.py]: it _should_ but just in-case this hasn't been implemented correctly (and I know of a few places where this isn't implemented 100% correctly)

**Pull request merged**